pkg/apis/serving/v1/route_lifecycle_test.go

```diff
@@ -473,11 +473,37 @@ func TestCertificateNotReadyWithBubbledUpMessage(t *testing.T) {
 func TestCertificateProvisionFailed(t *testing.T) {
 	r := &RouteStatus{}
 	r.InitializeConditions()
-	r.MarkCertificateProvisionFailed("cert")
+	r.MarkCertificateProvisionFailed(&netv1alpha1.Certificate{})
 
 	apistest.CheckConditionFailed(r, RouteConditionCertificateProvisioned, t)
 }
 
+func TestCertificateFailedWithBubbledUpMessage(t *testing.T) {
```

Comment:
I don't think I get the difference between these two tests. We mark the certificate provisioning as failed in both cases. Could you please elaborate the idea behind it?

Reply:
As @gabo1208 said in https://github.com/knative/serving/issues/14530, the difference is `Status=Unkown` is a status in progress which means it's just `not ready` for now, and the `Status=False` is a final state with failure.